src/commands/generate/typeGenerator.ts

```diff
@@ -47,6 +53,37 @@ export function generateType(
     }
 }
 
+/**
+ * Generates a file of the following format:
+ * ```
+ *  export type ExampleAlias = string & {
+ *     __conjure_flavor_type?: "_flavor_ExampleAlias";
+ *  };
+ * ```
+ */
+const FLAVOR_TYPE_FIELD = "__conjure_type";
+export async function generateAlias(
+    definition: IAliasDefinition,
+    knownTypes: Map<string, ITypeDefinition>,
+    simpleAst: SimpleAst,
+): Promise<void> {
+    if (isFlavorizable(definition.alias)) {
+        const tsTypeVisitor = new TsReturnTypeVisitor(knownTypes, definition.typeName, false);
+        const fieldType = IType.visit(definition.alias, tsTypeVisitor);
+        const sourceFile = simpleAst.createSourceFile(definition.typeName);
+        const typeAlias = sourceFile.addTypeAlias({
+            isExported: true,
+            name: "I" + definition.typeName.name,
+            type: `${fieldType} & { ${FLAVOR_TYPE_FIELD}?: "${definition.typeName.name}" }`,
```

Comment:
TODO: discuss what we want as the value of the flavor type field (for now it's the type name without the I prefix)